.github/scripts/format_dbt_test_failures.py

```diff
@@ -0,0 +1,256 @@
+        to an openpyxl.Workbook sheet for the tests tracked by this group."""
+        fieldnames = self.fieldnames
+        return [
+            [row.get(fieldname) for fieldname in fieldnames]
+            for row in self._rows
+        ]
+
+
+# Type representing a mapping of sheet names to the tests contained therein
+FailedTestsByCategory = typing.Dict[str, FailedTestGroup]
+
+
+def main() -> None:
+    """Entrypoint to this script. Parses dbt test failures and writes a
+    workbook of test failures to the output path."""
+    try:
+        run_results_filepath = sys.argv[1]
+    except IndexError:
+        run_results_filepath = os.path.join("target", "run_results.json")
+
+    try:
+        manifest_filepath = sys.argv[2]
+    except IndexError:
+        manifest_filepath = os.path.join("target", "manifest.json")
+
+    try:
+        output_filepath = sys.argv[3]
+    except IndexError:
+        date_today = datetime.datetime.today().strftime("%Y-%m-%d")
+        output_filepath = f"qc_test_failures_{date_today}.xlsx"
+
+    with open(run_results_filepath) as run_results_fobj:
+        run_results = json.load(run_results_fobj)
+
+    with open(manifest_filepath) as manifest_fobj:
+        manifest = json.load(manifest_fobj)
+
+    failed_tests_by_category = get_failed_tests_by_category(
+        run_results, manifest
+    )
+    if not failed_tests_by_category:
+        raise ValueError(f"{run_results_filepath} contains no failed rows")
+
+    workbook = openpyxl.Workbook()
+    for sheet_name, failed_test_group in failed_tests_by_category.items():
+        add_sheet_to_workbook(workbook, sheet_name, failed_test_group)
+    workbook.save(output_filepath)
+
+
+def get_failed_tests_by_category(
+    run_results: typing.Dict, manifest: typing.Dict
+) -> FailedTestsByCategory:
+    """Given two artifacts from a `dbt test --store-failures` call (a
+    run_results.json file dict and a manifest.json file dict), generates a dict
+    where each key is a name of a sheet and each associated value is
+    a list of failed tests for that sheet."""
+    conn = pyathena.connect(
+        s3_staging_dir=AWS_ATHENA_S3_STAGING_DIR,
+        region_name="us-east-1",
+        cursor_class=pyathena.cursor.DictCursor,
+    )
+    cursor = conn.cursor()
+
+    failed_tests_by_category: FailedTestsByCategory = {}
+
+    for result in run_results["results"]:
+        if result["status"] == "fail":
+            unique_id = result["unique_id"]
+            # Unique ID format is:
+            # test.athena.<test_name>.<hash>
+            test_name = unique_id.split(".")[2]
+
+            node = manifest["nodes"].get(unique_id)
+            if node is None:
+                raise ValueError(
+                    f"Missing dbt manifest node with id {unique_id}"
+                )
+
+            meta = node.get("meta", {})
+            category = meta.get("category", DEFAULT_TEST_CATEGORY)
+            test_description = meta.get("description")
```

Comment:
**suggestion:** Two other fields we may want to add here:

1. The actual test name, as shown in the `dbt test` output. We can hide this column in the worksheet if needed, but it's useful to have in there.
2. A link to the dbt docs page for the test, which shows the exact SQL definition.